ui/src/lib/store/index.tsx

```diff
@@ -27,7 +27,14 @@ export type Pod = {
   isSyncing?: boolean;
   children: { id: string; type: string }[];
   parent: string;
-  result?: { html?: string; text?: string; count: number; image?: string };
+  result?: {
+    type?: string;
+    html?: string;
+    text?: string;
+    count: number;
+    image?: string;
+  }[];
```

Comment:
I am not confident about changing this. It is a breaking change, the old values in the DB need migration to work with the new code.

If we really want to change this, we need to supply a DB migration script or procedure/function.

I think the issue you are trying to fix is:
1. the order of stderr
2. being able to display multiple images (at the end, not to be mixed with stdout/stderr streams)

I think you can fix (1) without introducing this schema change. For 2, I'd suggest skipping it for now, it's not crucial.

Reply:
Also, using an array to store streams doesn't sound like a good idea. The stream can come at any granularity, e.g., commonly line-by-line using because `\n` flushes stream in most languages, or users may call `flush()` manually.

I believe "stream" is supposed to be concatenated together upon receiving.

Reply:
In summary, I suggest fixing only issue 1 here, with minimal change so that we don't have to worry about migration. Forget about 2, it's not important.

Reply:
It's better to open another PR to fix issue 1, and leave this PR for future reference.

Reply:
> Also, using an array to store streams doesn't sound like a good idea. The stream can come at any granularity, e.g., commonly line-by-line using because `\n` flushes stream in most languages, or users may call `flush()` manually.
> 
> I believe "stream" is supposed to be concatenated together upon receiving.

So the change here is put all the returned messages from kernel in an array, rather than manually separating off each message and concatenating the `text` field. Ipynb kernel would decide how to concatenate each line's execution.

Reply:
> I am not confident about changing this. It is a breaking change, the old values in the DB need migration to work with the new code.
> 
> If we really want to change this, we need to supply a DB migration script or procedure/function.
> 
> I think the issue you are trying to fix is:
> 
> 1. the order of stderr
> 2. being able to display multiple images (at the end, not to be mixed with stdout/stderr streams)
> 
> I think you can fix (1) without introducing this schema change. For 2, I'd suggest skipping it for now, it's not crucial.

IIUC, the JSON format change will not render the `result` filed correctly in the existing repos, in that case, will a re-execution of each pod overwrite the result field?

Reply:
Yes, you are right, the re-execution will fix the result field. The only thing break is the existing result, which might not be that important to do a migration for it, especially at this early release point.